perf: Throttle UX features aggresively on large tables

ankush opened the pull request:
When table size exceeds a certain threshold (100K rows), the following behavior will change automatically:

- filters will not default to `like`, users can pick it still if they want to. 
- listview auto-refresh is debounced by 15s instead of 2s (default)


More can be added as we discover them, fixing all those problems in one shot is not possible.

sagarvora commented:
let's add a new operator "starts with" and make it default instead.

the alternative is tough to explain as a design decision

ankush commented:
I was thinking of making it configurable on docfield too. "Default comparison operator".

This is only a problem on large tables, perhaps we can switch default based on some heuristic.

ankush committed:
feat: estimate table size

ankush commented:
New solution:

- `get_count` will also return an estimated count of the entire table. 
- if count exceeds certain threshold, a lot of UX features will be automatically disabled or heavily throttled now. 

IMO 100K rows is a good threshold.

ankush committed:
feat: share table size heuristic with meta

ankush committed:
fix: don't apply "like" default on large tables

ankush committed:
fix: throttle list view refreshes for longer time

- 2s on small tables
- 15s on large tables

I don't want a huge range of numbers for this, this seems like a sane choice as
of now.
